src/arcaflow_plugin_sdk/atp.py

```diff
@@ -247,49 +319,53 @@ def read_hello(self) -> HelloMessage:
         message = self.decoder.decode()
         return _HELLO_MESSAGE_SCHEMA.unserialize(message)
 
-    def start_work(self, step_id: str, input_data: any):
+    def start_work(self, run_id: str, step_id: str, config: any):
         """
         After the Hello message has been read, this function starts work in a plugin with the specified data.
         """
-        self.encoder.encode(
+        self.send_runtime_message(
+            MessageType.WORK_START,
+            run_id,
             {
                 "id": step_id,
-                "config": input_data,
+                "config": config,
             }
         )
-        self.stdin.flush()
 
-    def send_signal(self, step_id: str, signal_id: str, input_data: any):
+    def send_signal(self, run_id: str, signal_id: str, input_data: any):
         """
         This function sends any signals to the plugin.
         """
-        self.send_runtime_message(MessageType.SIGNAL, {
-                "step_id": step_id,
+        self.send_runtime_message(
+            MessageType.SIGNAL,
+            run_id,
+        {
                 "signal_id": signal_id,
                 "data": input_data,
             }
         )
 
     def send_client_done(self):
-        self.send_runtime_message(MessageType.CLIENT_DONE, {})
+        self.send_runtime_message(MessageType.CLIENT_DONE, "", "")
```

Comment:
Granted, the type of the third parameter of `send_runtime_message` is `any`, but all the previous calls have sent `dict`'s...it seems like settling on a particular type would be good.  (If you really want to send nothing and don't like `{}`, how about `None`?)